### PR TITLE
Remove vertx circuit breaker

### PR DESCRIPTION
**Pull request by pierDipi**
Signed-off-by: Pierangelo Di Pilato <pierdipi@redhat.com>

### CODE REVIEW

**Comment by pierDipi** [comment]
Usual flake in 410-e2e-aws-ocp-410

**Comment by lberk** [comment]
/test 410-e2e-aws-ocp-410
--


<br class="Apple-interchange-newline">

File: data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/http/WebClientCloudEventSender.java [hunk 1]
```diff
@@ -78,41 +90,58 @@ public WebClientCloudEventSender(final WebClient client,
...
+      },
+      // Retry and failures
+      cause -> {
+        if (retryCounter+1 > egress.getRetry()) {
+          return Future.failedFuture(cause);
+        }
+
+        Promise<HttpResponse<Buffer>> r = Promise.promise();
+        final var delay = retryPolicyFunc.apply(retryCounter + 1);
+        vertx.setTimer(delay, v -> send(event, retryCounter + 1).onComplete(r));
```
Comment (matzew) [inline data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/http/WebClientCloudEventSender.java:141]: Do we need to take the timeout into account here ?

**Comment by matzew** [comment]
I assume this is a temporary fix, to work around the OOM issue?

(asking since we still have references to the breaker and its options)

File: data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/http/WebClientCloudEventSender.java (hunk 1 above)
Comment (pierDipi) [reply data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/http/WebClientCloudEventSender.java:141]: Timeout is set on the request itself, see line 161

**Comment by pierDipi** [comment]
> I assume this is a temporary fix, to work around the OOM issue?
> 
> (asking since we still have references to the breaker and its options)

Yes, we can decide if we want to drop it completely later and it was also to keep the diff smaller.

File: data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/http/WebClientCloudEventSender.java (hunk 1 above)
Comment (matzew) [reply data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/http/WebClientCloudEventSender.java:141]: yeah, via the breaker options, that;s why I added the other comment :-)

**Review by matzew — APPROVED** [review]
/lgtm
/approve

File: data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/http/WebClientCloudEventSender.java (hunk 1 above)
Comment (pierDipi) [reply data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/http/WebClientCloudEventSender.java:141]: ah ok, so yes, we're still configuring the CB options

**Pull request merged**